Source/AssistantV1/Assistant.swift

```diff
@@ -83,24 +83,27 @@ public class Assistant {
     }
 
     /**
-     If the response or data represents an error returned by the Watson Assistant service,
-     then return NSError with information about the error that occured. Otherwise, return nil.
+     Use the HTTP response and data received by the Watson Assistant service to extract
+     information about the error that occurred.
 
-     - parameter data: Raw data returned from the service that may represent an error.
-     - parameter response: the URL response returned from the service.
+     - parameter data: Raw data returned by the service that may represent an error.
+     - parameter response: the URL response returned by the service.
      */
-    func errorResponseDecoder(data: Data, response: HTTPURLResponse) -> Error {
+    func errorResponseDecoder(data: Data, response: HTTPURLResponse) -> RestError {
+
+        let statusCode = response.statusCode
+        var errorMessage: String?
+        var metadata = [String: JSON]()
 
-        let code = response.statusCode
         do {
             let json = try JSONDecoder().decode([String: JSON].self, from: data)
-            var userInfo: [String: Any] = [:]
             if case let .some(.string(message)) = json["error"] {
-                userInfo[NSLocalizedDescriptionKey] = message
+                errorMessage = message
+                metadata["error"] = JSON.string(message)
```

Comment:
I would like to get your opinion on this @mkistler. `RestError.http` has both a `message` and a `metadata` field. Any time there is a `message`, I also put that value in the `metadata` field. That is redundant, so I'm thinking of passing `nil` to the metadata field instead if there is only one value being extracted from the response JSON.

Reply:
I agree that we don't need to set the message in `metadata` if it is already stored in the separate `message` field.